src/server/scripts/Northrend/Ulduar/Ulduar/boss_auriaya.cpp

```diff
@@ -89,518 +107,605 @@ enum AuriayaYells
 enum AuriayaActions
 {
     ACTION_CRAZY_CAT_LADY                        = 0,
-    ACTION_RESPAWN_DEFENDER
+    ACTION_DEFENDER_DIED
 };
 
-#define SENTRY_NUMBER                            RAID_MODE<uint8>(2, 4)
-
-enum Mis
+enum Misc
 {
     DATA_NINE_LIVES                             = 30763077,
-    DATA_CRAZY_CAT_LADY                         = 30063007
+    DATA_CRAZY_CAT_LADY                         = 30063007,
+    PHASE_NONE                                  = 1,
+    PHASE_COMBAT                                = 2,
+    SUMMON_GROUP_10_MAN                         = 1,
+    SUMMON_GROUP_25_MAN                         = 2
 };
 
-class boss_auriaya : public CreatureScript
+class CatsTargetSelector : public std::unary_function<Unit*, bool>
```

Comment:
std::unary_function is deprecated and removed in c++17

Reply:
what should be used in this case?

Reply:
just remove the inheritance if lambdas are not an option here